### DISCUSSION TITLE
Elastic APM Agent with Open telemetry plugin

### DISCUSSION THREAD

**Discussion by AyushCarb1** [open-telemetry/opentelemetry-java-instrumentation#9881] [Q&A] [answered]
Hi Team,

We have referred Opentelemetry Bridge which says manual instrumentation is required to use Opentelemetry APIs with Elastic APM Agent.
Reference : https://www.elastic.co/guide/en/apm/agent/java/current/opentelemetry-bridge.html

We wanted to check if there is any way without manual instrumentation that we can use OpenTelemetry plugin with Elastic APM agent .

**Comment by laurit** [comment]
@jackshirazi could you take a look

**Comment by jackshirazi** [comment, marked as answer]
Hi @AyushCarb1,

I can answer for Elastic.

The Elastic APM agent automatically allows you to use Opentelemetry APIs, eg you can use (as shown in the example from that reference you linked)
```
Span custom = tracer.spanBuilder("my custom span").startSpan();
```
in your code where Span is an Opentelemetry Span, and the Elastic APM agent will correctly handle the span

If your question is that you have an Opentelemetry extension or instrumentation plugin and you would like to use that with the Elastic agent, that is not supported at this time. We (Elastic) are working on converging the Elastic APM agent to the Opentelemetry agent and so that will become possible at some point in the future

**Comment by baole229801** [comment]
Hi Mr. @jackshirazi , for example, if I only want to use the r2dbc instrumentation of OpenTelemetry with Elastic agent, what should be the approach ? Thank you so much.
